### DISCUSSION TITLE
Question to synthesized GeneratedNames for yield statement

### DISCUSSION THREAD

**Discussion by daveMueller** [dotnet/roslyn#68664] [General]
Hi,

I would need some clarification on how the synthesized names are generated for yield statements. We have a bug in our
coverage tool [https://github.com/coverlet-coverage/coverlet/issues/1484](https://github.com/coverlet-coverage/coverlet/issues/1484) because of this.

To be more precise, when having a simple class like the following
```csharp
public class PolicyNames
{
    public static IEnumerable<string> All()
    {
        yield return "1";
    }
}
```
the compiler is adding a nested type called `<All>d__0`
```csharp
.class nested private auto ansi sealed beforefieldinit '<All>d__0'
```

So now when I add a simple field like the following
```csharp
public class PolicyNames
{
    private string Read;

    public static IEnumerable<string> All()
    {
        yield return "1";
    }
}
```

the compiler is adding nested type called `<All>d__1`
```csharp
.class nested private auto ansi sealed beforefieldinit '<All>d__1'
```

Our algorithem currently only counts methods for the ordinal number and I guess this is the problem. What exactly (fields, properties, ...) do I need to consider when computing this generated name?

**Comment by jcouv** [comment]
Sorry, but the generation of closures is rather complicated and is an implementation detail (subject to change). Someone would have to dig into the compiler code, so we're unlikely to be able to help with this. I'll move the issue to a discussion

**Comment by MillKaDe** [comment]
Would it be possible to lookup the compiler generated name using reflection ?

[Code at SharpLab](https://sharplab.io/#v2:C4LgTgrgdgNAJiA1AHwAICYCMBYAUKgBgAJVMAWAbj0JMwDZaA6AYQHsoBnVgGwFMrceAG4BDMEQ68AjgEEiAXiIzGM7tyIAKAJQDR44AE8ADrzmLJsxgHFewACrHemncLFEoIgLamFRQyeUAOS9+PAB1MABLYF4AGUioJw0AEgAiGRAiAG8PbxkAX1SdIgB6EqIAHlVuAD44AH16gjxXcQsAIV92lTVnXTd/Xk7zaW6be0c+1vcQ4b9HbuDvAQjouISktPbMnNnC4rLK6rrGnEF8AGYSdCVsvCISK9IGUguK0gIapTVtbKJ7h60TAkADsRFSmFSAge+TwsPOqCeN06WQBHyIACVeCI4NDHrQXpg3h8vtVfll/rhAUDQeDIXj4fCgA==)

```c#
using System;
using System.Collections.Generic;
using static System.Console;

var seqA = A.All ();
var typeA = seqA.GetType ();
var nameA = typeA.Name;
WriteLine ($"A: {nameA}"); // <All>d__0

var seqB = B.All ();
var typeB = seqB.GetType ();
var nameB = typeB.Name;
WriteLine ($"B: {nameB}"); // <All>d__1

public class A {
  public static IEnumerable<string> All() { 
    yield return "1";
  }
}

public class B {
  string Read;
  public static IEnumerable<string> All() { 
    yield return "1";
  }
}
```

Greetings from KA to RA ..
